### PR TITLE
gha: use oidc

### PR DESCRIPTION
**Pull request by andrewhsu**
## Description

jira: [PESDLC-1729]

use [oidc](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services) to retrieve secrets

also replace `curl` api calls in `backport.yml` with GitHub CLI [gh](https://cli.github.com/) which comes [installed](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md) on github runners.

no functionality change.

## Page previews

N/A

## Checks

- [ ] New feature
- [ ] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

[PESDLC-1729]: https://redpandadata.atlassian.net/browse/PESDLC-1729?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/backport.yml [hunk 1]
```diff
@@ -30,25 +30,17 @@ jobs:
...
-
-          if [[ $? -ne 0 ]]; then
-            echo "Failed to fetch labels from GitHub API."
-            exit 1
-          fi
-
-          BACKPORT_LABELS=$(echo "$LABELS_JSON" | jq -r 'if type == "object" then [.] else . end | map(select(.name | startswith("backport to")))[] | .name')
-          if [[ -z "$BACKPORT_LABELS" ]]; then
+          # concat labels into comman-separated string, e.g. "bug,backport to all versions,remediation"
+          PR_LABELS="${{ join(github.event.pull_request.labels.*.name, ',') }}"
```
Comment (ivotron) [inline .github/workflows/backport.yml:24]: nice!

**Review by ivotron — APPROVED** [review]
i see many of these don't run in PRs, so i guess we'll need to keep an eye on them when they get executed

**Comment by andrewhsu** [comment]
rebased with the tip of `main`. no other changes.

**Review by JakeSCahill — APPROVED** [review]
Thanks for the improvements!

**Pull request merged**